heron/instance/src/java/com/twitter/heron/instance/bolt/BoltInstance.java

```diff
@@ -96,6 +97,14 @@ public BoltInstance(PhysicalPlanHelper helper,
   }
 
   @Override
+  public void update(PhysicalPlanHelper physicalPlanHelper) {
+    if (bolt instanceof IUpdatable) {
+      ((IUpdatable) bolt).update(physicalPlanHelper.getTopologyContext());
```

Comment:
Good point - this needs to be addressed. If the function execution exceeds beyond a certain time - some action needs to be taken.